### ISSUE TITLE
[Bug] Completed activity for which cancellation was requested causes duplicate `ResolveActivity` activation job

### ISSUE THREAD

**Issue by bergundy** [temporalio/sdk-core#110]
**Describe the bug**
When a Workflow requests to cancel an Activity, Core uses `TRY_CANCEL` mode and immediately activates the workflow with a `ResolveActivity` activation where the result is `canceled`.
If the activity completes before it heartbeats again core will resolve the activity again, this time with result set to `completed`.
The node SDK will throw an error saying that there's no pending completion for that activity.

**To Reproduce**
Run the `cancel-http-request` workflow in the node sdk and set the activity's `heartbeatTimeout` to something high like `3 minutes`.

**Expected behavior**
Core should not resend an `ActivityResolve` activation job.

**Screenshots/Terminal ouput**
```
Caught error while worker was running Error: No completion for taskSeq 0
```

**Comment by bergundy** [comment]
Run the the test with this command:
```shell
RUN_INTEGRATION_TESTS=1 npx ava packages/test/lib/test-integration.js -m "cancel-http-request don't completeOnActivityCancellation" -T 100s
```

**Comment by bergundy** [comment]
@Sushisource was this fixed by #112?

**Comment by Sushisource** [comment]
Indeed it was, thanks for the reminder.
